src/actions/query-keyspaces.ts

```diff
@@ -3,23 +3,38 @@
 import { actionClient } from "@scylla-studio/lib/safe-actions";
 import { z } from "zod";
 
-import { BatchStatement, Cluster, Query } from "@lambda-group/scylladb";
+import { Cluster } from "@lambda-group/scylladb";
 import { parseKeyspaces } from "@scylla-studio/lib/cql-parser/parser";
 
 
 
 export const queryKeyspaceAction = actionClient
-  .schema(z.object({}))
-  .action(async ({ parsedInput }) => {
-    const cluster = new Cluster({
-      nodes: ["localhost:9042"],
+    .schema(
+        z.object({
+            host: z.string().min(1, "Host is required"),
+            port: z.number().min(1, "Port is required"),
+            username: z.string().nullable(),
+            password: z.string().nullable(),
+        })
+    )
+    .action(async ({ parsedInput }) => {
+        const { host, port, username, password } = parsedInput;
+
+        const clusterConfig: any = {
+            nodes: [`${host}:${port}`],
+        };
+
+        if (username && password) {
+            clusterConfig.auth = {
+                username,
+                password,
+            };
+        }
+
+        const session = await new Cluster(clusterConfig).connect();
```

Comment:
That's not the best idea, here you are giving up the reference to the Cluster Class itself, we don't know how this is going to be garbage collected in the server as the client action lifespan is abstract away, the best approach here is to actually keep as it was before.